src/main/java/net/fabricmc/loom/task/AbstractRunTask.java

```
@@ -135,7 +145,74 @@ public void exec() {
 		setWorkingDir(new File(getProjectDir().get(), getInternalRunDir().get()));
 		environment(getInternalEnvironmentVars().get());
 
-		super.exec();
+		// Wrap with XVFB if enabled and on Linux
+		if (shouldUseXvfb()) {
+			LOGGER.info("Using XVFB for headless client execution");
+			execWithXvfb();
+		} else {
+			super.exec();
+		}
+	}
+
+	private boolean shouldUseXvfb() {
+		return getUseXvfb().get() && Platform.CURRENT.getOperatingSystem().isLinux();
```

If it gets set to true on another platform just try it, and let it fail.